client/docs/querying-the-api.md

```diff
@@ -24,19 +24,32 @@ If there is no Composable for the API endpoint you are using, try using a (Pinia
 
 ### Direct API Calls
 
--   If the type of data you are querying should not be cached, or you just need to update or create new data, you can use the API directly. Make sure to use the **Fetcher** (see below) instead of Axios, as it provides a type-safe interface to the API along with some extra benefits.
+-   If the type of data you are querying should not be cached, or you just need to update or create new data, you can use the API directly. Make sure to use the **GalaxyApi client** (see below) instead of Axios, as it provides a type-safe interface to the API along with some extra benefits.
 
-## 2. Prefer Fetcher over Axios (when possible)
+## 2. Prefer **GalaxyApi client** over Axios (when possible)
 
--   **Use Fetcher with OpenAPI Specs**: If there is an OpenAPI spec for the API endpoint you are using (in other words, there is a FastAPI route defined in Galaxy), always use the Fetcher. It will provide you with a type-safe interface to the API.
+-   **Use **GalaxyApi client** with OpenAPI Specs**: If there is an OpenAPI spec for the API endpoint you are using (in other words, there is a FastAPI route defined in Galaxy), always use the GalaxyApi client. It will provide you with a type-safe interface to the API.
 
 **Do**
 
 ```typescript
-import { fetcher } from "@/api/schema";
-const datasetsFetcher = fetcher.path("/api/dataset/{id}").method("get").create();
+import { GalaxyApi } from "@/api";
+
+const { data, error } = await GalaxyApi().GET("/api/datasets/{dataset_id}", {
+    params: {
+        path: {
+            dataset_id: datasetId,
+        },
+        query: { view: "detailed" },
+    },
+});
+
+if (error) {
+    // Handle error here and return otherwise `data` will be undefined
+    return;
```

Comment:
I get that's not the intention, but if someone takes this example literally that is exactly what we should avoid 😆.
Maybe we can have a real example here ?

Reply:
You are right, lazy doc writing... is worse than no doc at all sometimes :sweat_smile:

Reply:
Is something like https://github.com/galaxyproject/galaxy/pull/18532/commits/1ffd38242bb4926d2a0456dd05cdedd078520849 slightly better?